src/wasm-interpreter.h

```diff
@@ -3494,6 +3499,83 @@ class ModuleRunnerBase : public ExpressionRunner<SubType> {
     }
     return {};
   }
+  Flow visitArrayNewSeg(ArrayNewSeg* curr) {
+    NOTE_ENTER("ArrayNewSeg");
+    auto offsetFlow = self()->visit(curr->offset);
+    if (offsetFlow.breaking()) {
+      return offsetFlow;
+    }
+    auto sizeFlow = self()->visit(curr->size);
+    if (sizeFlow.breaking()) {
+      return sizeFlow;
+    }
+
+    auto offset = offsetFlow.getSingleValue().geti32();
+    auto size = sizeFlow.getSingleValue().geti32();
+
+    auto heapType = curr->type.getHeapType();
+    const auto& element = heapType.getArray().element;
+    auto elemType = heapType.getArray().element.type;
+
+    Literals contents;
+    contents.reserve(size);
+
+    switch (curr->op) {
+      case NewData: {
+        assert(curr->segment < wasm.dataSegments.size());
+        assert(elemType.isNumber());
+        const auto& seg = *wasm.dataSegments[curr->segment];
+        auto elemBytes =
```

Comment:
Maybe `Field` could have `getByteSize` that does this?

Reply:
It turns out it already does!